storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2529,7 +2529,7 @@ public void submitTopologyWithOpts(String topoName, String uploadedJarLocation,
                 }
             }
             validateTopologyWorkerMaxHeapSizeConfigs(topoConf, topology);
-            Utils.validateTopologyBlobStoreMap(topoConf, Sets.newHashSet(blobStore.listKeys()));
+            Utils.validateTopologyBlobStoreMap(topoConf);
```

Comment:
Maybe better to add method which contains NimbusBlobStore as parameter to avoid creating NimbusBlobStore in Nimbus.